src/loaders/lottie/tvgLottieBuilder.cpp

```diff
@@ -371,9 +373,65 @@ static void _repeat(LottieGroup* parent, unique_ptr<Shape> path, RenderContext*
 }
 
 
+static void _appendSharpRect(Shape* shape, float x, float y, float w, float h, float r)
+{
+    constexpr int commandsCnt = 5;
+    PathCommand commands[commandsCnt] = {
+        PathCommand::MoveTo, PathCommand::LineTo, PathCommand::LineTo,
+        PathCommand::LineTo, PathCommand::Close
+    };
+
+    constexpr int pointsCnt = 4;
+    Point points[pointsCnt] = {{x + w, y}, {x + w, y + h}, {x, y + h}, {x, y}};
+
+    shape->appendPath(commands, commandsCnt, points, pointsCnt);
+}
+
+static void _appendRoundedRect(Shape* shape, float x, float y, float w, float h, float r)
+{
+    constexpr int commandsCnt = 10;
+    PathCommand commands[commandsCnt] = {
```

Comment:
this command list style would be good to apply in tvgShape.cpp as well.

Reply:
@hermet  I'm not sure if I understand... in 'appendRect' we use functions from pImpl like 'moveTo', 'cubicTo' etc. would you like to replace them with 'append' from pImpl?